### ISSUE TITLE
does autoscale_lr take affect?

### ISSUE THREAD

**Issue by iamRyanChia** [open-mmlab/mmdetection#3079]
https://github.com/open-mmlab/mmdetection/blob/bf5c7185e8153af1d9aadd79e472d889594198a4/tools/train.py#L92

when running with `./tools/dist_train.sh config 4 --autoscale_lr`, the lr is not scaled coorectly, unless i specify the gpu-ids or gpus like `./tools/dist_train.sh config 4 --autoscale_lr --gpus 4`, it is unnatural.

**Comment by hellock** [comment]
Thanks for reporting! This argument is risky for distributed training and we decide to remove the misleading arg in #3080.

**Comment by FreyWang** [comment]
I have changed it into:
`cfg.optimizer['lr'] = cfg.optimizer['lr'] * cfg.data['samples_per_gpu'] * torch.cuda.device_count() / 16`

**Comment by hellock** [comment]
> I have changed it into:
> `cfg.optimizer['lr'] = cfg.optimizer['lr'] * cfg.data['samples_per_gpu'] * torch.cuda.device_count() / 16`

Sometimes users only use part of GPUs on the server, e.g., 4 out of 8. `torch.cuda.device_count()` is not safe.

**Comment by FreyWang** [comment]
> > I have changed it into:
> > `cfg.optimizer['lr'] = cfg.optimizer['lr'] * cfg.data['samples_per_gpu'] * torch.cuda.device_count() / 16`
> 
> Sometimes users only use part of GPUs on the server, e.g., 4 out of 8. `torch.cuda.device_count()` is not safe.

yes it is true, cause I always use `CUDA_VISIBLE_DEVICES` to assign gpu 😁

**Comment by hellock** [comment]
That's good. However the usage cases are various thus we just remove this argument to prevent bugs and always make users specify the learning rate.
